synapse/http/federation/srv_resolver.py

```diff
@@ -109,7 +109,7 @@ class SrvResolver:
 
     def __init__(
         self,
-        dns_client=client,
+        dns_client: Any = client,
```

Comment:
Can this be more specific? I think it has to implement something from `twisted.names`?

Reply:
In this situation `twisted.names.client` is a module(!) without a convenient type. In sydent I changed this class to accept just the function from twisted that we need, see here.

https://github.com/matrix-org/sydent/blob/7c76e2fcec8691a0efe7e9d90073897b628dbae7/sydent/http/srvresolver.py#L118

Reply:
Ah! I remember doing *something* about this, guess it was in sydent. Thank you for the reminder! 👍